### PR TITLE
Action for updating symbols

### PR DESCRIPTION
**Pull request by EsoFromTheEast**
Create an Action for updating symbols

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.11
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        if [ -f update/requirements.txt ]; then pip install -r update/requirements.txt; fi
```
Comment (rsheeter) [inline .github/workflows/main.yml:21]: you don't need the if do you? requirements.txt should always be htere?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Run Update Script
+      working-directory: ./update
+      run: |
+        python update_symbols.py
+    - name: commit
+      run: |
+        git config user.name "GitHub Actions Bot"
+        git config user.email "<>"
+        git add .
+        git commit -m "Update Symbols"
```
Comment (rsheeter) [inline .github/workflows/main.yml:31]: I think you can stick name/email in the commit? - `git commit --author "blah <blah>" type of deal?

**Review by rsheeter — APPROVED** [review]
LG assuming you've tested this and it seems to work.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  update-symbols:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
```
Comment (2bndy5) [inline .github/workflows/main.yml:15]: You will find that the versions specified here will trigger warnings in the CI summary related to using an old version of node.js. The fix for this is to use the latest version:
```diff
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Python
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
```
Sorry, I'm a little late to give this insight.

**Comment by nasernadery** [comment]
باصفا باشید

**Comment by nasernadery** [comment]
ایران گیلان لنگرود پل خشتی لنگرود
naser nadery
nn2503060@gmail.com

File: .github/workflows/main.yml (hunk 2 above)
Comment (EsoFromTheEast) [reply .github/workflows/main.yml:31]: done

File: .github/workflows/main.yml (hunk 1 above)
Comment (EsoFromTheEast) [reply .github/workflows/main.yml:21]: done

File: .github/workflows/main.yml (hunk 3 above)
Comment (EsoFromTheEast) [reply .github/workflows/main.yml:15]: done

File: .github/workflows/main.yml (hunk 1 above)
Comment (nasernadery) [reply .github/workflows/main.yml:21]: nader nasery nadernasery184.8585@gmail.com

**Pull request merged**